### PR TITLE
Clean up is_constant is_input is_output in Tensor

### PR DESCRIPTION
**Pull request by yxlao**
- `is_persistent` is always `false` in our current implementation, so it can be removed
- `is_constant` `is_input` and `is_output` is only consumed in `liveness.cpp`, so we could compute them there directly
- The general idea is that, a node should have info about whether its output tensor(s) is(are) persistent or not, which is related to the node type. The tensors shouldn't need to have that info

### CODE REVIEW

File: src/ngraph/pass/memory_visualize.cpp [hunk 1]
```diff
@@ -246,25 +243,17 @@ int pass::MemoryVisualize::compute_op_weight(const shared_ptr<Node> exop)
     int mass = 0;
     // for input_decl in exop.input_decls:
```
Comment (ransford2011) [inline src/ngraph/pass/memory_visualize.cpp]: This looks like Python++.  I'm not seeing the value in keeping it around.  This should be removed.

**Comment by yxlao** [comment]
TESTNOW

**Comment by yxlao** [comment]
TESTNOW

**Comment by yxlao** [comment]
TESTNOW

**Comment by DawnStone** [comment]
TESTNOW

**Comment by DawnStone** [comment]
TESTNOW

**Comment by DawnStone** [comment]
TESTNOW

**Comment by yxlao** [comment]
@Krovatkin fixed

**Pull request merged**